tasks/main.yml

```
@@ -17,26 +12,237 @@
+  register: __mssql_variables_evaluation
+
+- name: Set up MSSQL
+  command: /opt/mssql/bin/mssql-conf -n setup
+  environment:
+    - ACCEPT_EULA: "Y"
+    - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+    - MSSQL_PID: "{{ mssql_edition }}"
+  when: not __mssql_is_setup
+  register: __mssql_conf_setup
+
+- name: Configure the Microsoft SQL Server Tools repository
+  yum_repository:
+    name: packages-microsoft-com-prod
+    description: Microsoft SQL Server Tools
+    baseurl: "{{ __mssql_client_repository }}"
+    gpgcheck: true
+
+- name: Ensure that SQL Server client tools are installed
+  package:
+    name: "{{ __mssql_client_packages }}"
+    state: present
+  environment:
+    - ACCEPT_EULA: Y
+
+- name: Set a new password for the MSSQL sa user
+  when:
+    - __mssql_conf_setup is skipped
+    - mssql_password is defined
+  block:
+    - name: Log in using the set password
+      include_tasks: verify_password.yml
+
+    - name: Ensure that the mssql-server service is stopped
+      service:
+        name: mssql-server
+        state: stopped
+      when: __mssql_password_query is failed
+      notify: Restart the mssql-server service
+
+    - name: Change the password of the sa user
+      command: /opt/mssql/bin/mssql-conf set-sa-password
+      environment:
+        - MSSQL_SA_PASSWORD: "{{ mssql_password }}"
+      when: __mssql_password_query is failed
+      notify: Restart the mssql-server service
+
+- name: Set a new edition for MSSQL
+  when:
+    - __mssql_conf_setup is skipped
+    - mssql_edition is defined
+  block:
+    - name: Check if the set edition matches the existing edition
+      shell: |
+        errorlog_edition="$(grep -oi '{{ mssql_edition }} edition' \
+        {{ __mssql_errorlog.stdout }})"
+        if [ -z "${errorlog_edition}" ]; then
+          edition_matches=false
+        else
+          edition_matches=true
+        fi
+        echo "${edition_matches}"
+      register: __mssql_edition_matches
+      ignore_errors: true
```

do you need to ignore errors here?